src/main/java/com/project/service/CommentServiceImpl.java

```diff
@@ -0,0 +1,82 @@
+package com.project.service;
+
+import com.project.exception.CommentNotFoundException;
+import com.project.exception.ApplicationNotFoundException;
+import com.project.exception.UserNotFoundException;
+import com.project.model.Comment;
+import com.project.model.request.CommentRequest;
+import com.project.model.response.CommentResponse;
+import com.project.repository.CommentRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class CommentServiceImpl implements CommentService {
+
+    private final CommentRepository commentRepository;
+    private final ApplicationService applicationService;
+    private final UserService userService;
+
+    @Autowired
+    public CommentServiceImpl(CommentRepository commentRepository, ApplicationService applicationService, UserService userService) {
+        this.commentRepository = commentRepository;
+        this.applicationService = applicationService;
+        this.userService = userService;
+    }
+
+    @Override
+    public List<CommentResponse> getAll(Long appId) {
+        List<Comment> comments = new ArrayList<>(commentRepository.findAll());
+        return comments.stream().filter(comment -> comment.getApplication().getId().equals(appId)).map(CommentResponse::new)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public CommentResponse getById(Long id) throws CommentNotFoundException {
+        Comment comment = commentRepository.findById(id).orElseThrow(() -> new CommentNotFoundException(id));
+        return new CommentResponse(comment);
+    }
+
+    @Override
+    public List<CommentResponse> getApplicantVisibleComments(Long appId) {
+        List<CommentResponse> comments = getAll(appId);
+        return comments.stream().filter(CommentResponse::isVisibleToApplicant).collect(Collectors.toList());
+    }
+
+    @Override
+    public CommentResponse addAdminComment(CommentRequest comment, Long appId)
+            throws ApplicationNotFoundException, UserNotFoundException {
+        DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
```

Comment:
You use the same `DateFormat` multiple times. Maybe you could have just only one instance of it?